util/stream.go

```diff
@@ -64,9 +67,8 @@ func NewMessageStream(conn net.Conn, parser Parser) *MessageStream {
 	go m.outbound()
 	go m.inbound()
 
-	for i := 0; i < 25; i++ {
```

Comment:
@jojimt: I've done some stress testing with Hello messages to measure the impact of these changes. The code needs about 3.5 seconds to process 5000000 messages with the function we use in ofnet to parse the messages (https://github.com/contiv/ofnet/blob/master/ofctrl/ofctrl.go#L174). Do we happen to know if we need to process that many messages in such a short period of time in production?

The old code (without any of the atomic fixes and any of these locking improvements) took about 3.2-3.4 seconds. The new code with all the fixes takes about 3.3-3.7 seconds to process 5000000 messages. That's between 640 nanoseconds and 760 nanoseconds per message.

The parser function appears to be fast enough to not matter if we process in parallel or sequentially.

Do you happen to have any ideas or any other tests in mind? Please let me know if you have any other suggestions.

I've pushed a few changes to remove the use of one channel and use sync.Pool.